job_creator.php

```diff
@@ -119,6 +116,10 @@ private function getCleanedBranch(): string
         }
         // e.g. 4.10-release
         $branch = preg_replace('#^([0-9\.]+)-release$#', '$1', $branch);
+        // fallback to parent branch if availableable
+        if (!is_numeric($branch) && $this->parentBranch && is_numeric($this->parentBranch)) {
```

Comment:
What is parentBranch is a `-release` branch?

Reply:
Updated

Reply:
Just moving the `-release` branch regex to be below this won't work, because we still aren't setting `$branch` to `$this->parentBranch` in that even since `4.11-release` is not numeric.

Reply:
Good point, have updated and added unit test